crates/trigger-http/src/lib.rs

```diff
@@ -487,48 +471,71 @@ fn strip_forbidden_headers(req: &mut Request<Body>) {
 // modules is going to be different, so each executor must must use the info
 // in its standardized way (environment variables for the Wagi executor, and custom headers
 // for the Spin HTTP executor).
-const FULL_URL: &[&str] = &["SPIN_FULL_URL", "X_FULL_URL"];
-const PATH_INFO: &[&str] = &["SPIN_PATH_INFO", "PATH_INFO"];
-const MATCHED_ROUTE: &[&str] = &["SPIN_MATCHED_ROUTE", "X_MATCHED_ROUTE"];
-const COMPONENT_ROUTE: &[&str] = &["SPIN_COMPONENT_ROUTE", "X_COMPONENT_ROUTE"];
-const RAW_COMPONENT_ROUTE: &[&str] = &["SPIN_RAW_COMPONENT_ROUTE", "X_RAW_COMPONENT_ROUTE"];
-const BASE_PATH: &[&str] = &["SPIN_BASE_PATH", "X_BASE_PATH"];
-const CLIENT_ADDR: &[&str] = &["SPIN_CLIENT_ADDR", "X_CLIENT_ADDR"];
-
-pub(crate) fn compute_default_headers<'a>(
+const FULL_URL: [&str; 2] = ["SPIN_FULL_URL", "X_FULL_URL"];
+const PATH_INFO: [&str; 2] = ["SPIN_PATH_INFO", "PATH_INFO"];
+const MATCHED_ROUTE: [&str; 2] = ["SPIN_MATCHED_ROUTE", "X_MATCHED_ROUTE"];
+const COMPONENT_ROUTE: [&str; 2] = ["SPIN_COMPONENT_ROUTE", "X_COMPONENT_ROUTE"];
+const RAW_COMPONENT_ROUTE: [&str; 2] = ["SPIN_RAW_COMPONENT_ROUTE", "X_RAW_COMPONENT_ROUTE"];
+const BASE_PATH: [&str; 2] = ["SPIN_BASE_PATH", "X_BASE_PATH"];
+const CLIENT_ADDR: [&str; 2] = ["SPIN_CLIENT_ADDR", "X_CLIENT_ADDR"];
+
+pub(crate) fn compute_default_headers(
     uri: &Uri,
-    raw: &str,
     base: &str,
     host: &str,
+    route_match: &RouteMatch,
     client_addr: SocketAddr,
-) -> Result<Vec<(&'a [&'a str], String)>> {
+) -> Result<Vec<([Cow<'static, str>; 2], String)>> {
```

Comment:
I would return `String`s here. Avoiding allocations is nice and all but all these headers get rewritten a couple of times downstream anyway; probably not worth the extra complexity here.